compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -334,23 +334,61 @@ xmaxxminHelper(TR::Node* node, TR::CodeGenerator* cg, TR::InstOpCode::Mnemonic c
 
    TR::LabelSymbol* cFlowRegionStart = generateLabelSymbol(cg);
    TR::LabelSymbol* cFlowRegionEnd = generateLabelSymbol(cg);
+   //Label Symbol for handling NaN operands
+   TR::LabelSymbol* nanLabel = generateLabelSymbol(cg);
 
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionStart);
    cFlowRegionStart->setStartInternalControlFlow();
 
    generateRREInstruction(cg, compareRROp, node, lhsReg, rhsReg);
    generateS390BranchInstruction(cg, TR::InstOpCode::BRC, branchCond, node, cFlowRegionEnd);
+   //Branching to nanLabel for Condition Code 3 when operands are unordered
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_CC3, node, nanLabel);
 
    generateRREInstruction(cg, moveRROp, node, lhsReg, rhsReg);
+   //Branching to cFlowRegionEnd to jump over NaN case
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BRC, node, cFlowRegionEnd);
+   //Handle NaN case
+   generateS390LabelInstruction(cg, TR::InstOpCode::label, node, nanLabel);
 
-   TR::RegisterDependencyConditions* deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg);
+   TR::RegisterDependencyConditions* deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 4, cg);
 
    deps->addPostConditionIfNotAlreadyInserted(lhsReg, TR::RealRegister::AssignAny);
    deps->addPostConditionIfNotAlreadyInserted(rhsReg, TR::RealRegister::AssignAny);
+   //Cursor for memory reference
+   TR::Instruction * cursor;
 
+   //Handle NaN case for doubles
+   if (node->getOpCode().isDouble())
```

Comment:
Is this evaluator used for non floating type min max as well (integer , long , etc) ? 
I believe we pass in the move Op and compare Op so it will generate appropriate instructions for such cases.

Current change in this PR would have NaN case for all the types whereas Integer / Long do not have a NaN. 

Can you confirm? If this is true than, we need to refactor and add checks so that only Double and Float would generate NaN case.

Reply:
Thanks for confirming. I think you can refactor the single function to handle this scenario. You need to guard the NaN changes with isDouble/Float query. You can tackle the common condition code early on (CC 0/1/2) first and lastly you can generate code for NaN case)

Reply:
@sarwat12 You still need to work on this suggestion. Evaluator in current state will allocate a register to hold NaN value for Integral type min and max node as well. Code for NaN should only be generated for float and double.